src/Models/class-smart-link.php

```diff
@@ -196,8 +196,9 @@ public function __construct(
 	private function get_smart_link_object_by_uid( string $uid ): int {
 		$cache_key = self::get_uid_to_smart_link_cache_key( $uid );
 		$cached    = wp_cache_get( $cache_key, PARSELY_CACHE_GROUP );
-		if ( is_int( $cached ) && 0 !== $cached ) {
-			return $cached;
+
+		if ( is_numeric( $cached ) && 0 !== (int) $cached ) {
```

Comment:
Same here:

```suggestion
		if ( false !== $cached && is_numeric( $cached ) ) {
```

The `0 !== (int) $cached` check seems unnecessary here with a `false` check.

Reply:
I kept it in case there was a reason for it. I didn't dig further on why we're checking for a potential 0 value. Do you think it's irrelevant?

Reply:
That's a good point! Looking at the cache setting code, it appears we never will set `0` in this cache anyway:

https://github.com/Parsely/wp-parsely/blob/5cf945848bde41842cb5eaec6e4f367354f2decc/src/Models/class-smart-link.php#L355-L366

So I think the suggested change is safe to make.